Micro-optimization: only split `acct` into two Strings

postmodern opened the pull request:
Since `acct` is split by `@` and assigned to `username` and `domain`, we only need to split `acct` into two Strings.

postmodern committed:
Micro-optimization: only split `acct` into two Strings

* Since `acct` is split by `@` and assigned to `username` and `domain`, we only need to split `acct` into two Strings.

sunny commented on app/models/account_alias.rb:
```diff
@@ -29,7 +29,7 @@ def acct=(val)
 
   def pretty_acct
-    username, domain = acct.split('@')
+    username, domain = acct.split('@',2)
```
```suggestion
    username, domain = acct.split('@', 2)
```

postmodern replied:
Manually edited and force pushed.

ClearlyClaire commented:
Does that actually bring any benefit?

postmodern commented:
> Does that actually bring any benefit?

Yes, by about 50%.

```ruby
#!/usr/bin/env ruby
require 'benchmark'

Benchmark.bm do |b|
  n = 1_000_000

  account = 'foo@bar@baz@qux.com'

  b.report('String#split(\'@\')') do
    n.times do
      user, domain = account.split('@')
    end
  end

  b.report('String#split(\'@\',2)') do
    n.times do
      user, domain = account.split('@',2)
    end
  end
end
```

```
       user     system      total        real      
String#split('@')  0.671949   0.002190   0.674139 (  0.679637)
String#split('@',2)  0.436089   0.000990   0.437079 (  0.439453)
```

ClearlyClaire commented:
In this case, however, it is known to have at most one `@` because that method is only ever used on accts that are saved to the database and deemed valid as far as I can tell.

postmodern commented:
> In this case, however, it is known to have at most one `@` because that method is only ever used on accts that are saved to the database and deemed valid as far as I can tell.

Are you sure that is always going to be the case? Better to be explicit since in this case it is extracting only two components.

ClearlyClaire commented:
I'm confident that won't change anytime soon, and it's far from being a hot spot in the code path. But I see no downside to this either way.
